lib/web/apps.go

```diff
@@ -77,14 +77,22 @@ func (h *Handler) clusterAppsGet(w http.ResponseWriter, r *http.Request, p httpr
 	}, nil
 }
 
-type GetAppFQDNRequest resolveAppParams
+type GetAppFQDNRequest ResolveAppParams
 
 type GetAppFQDNResponse struct {
 	// FQDN is application FQDN.
 	FQDN string `json:"fqdn"`
 }
 
-type CreateAppSessionRequest resolveAppParams
+// CreateAppSessionRequest respresents the payload present when creating
+// application sessions.
+type CreateAppSessionRequest struct {
+	ResolveAppParams
+	// PreflightConnection when set the proxy preflight a connection to an
+	// AppServer, ensuring it can handle the request. If the preflight fails,
+	// the session is not created and it returns an error.
+	PreflightConnection bool `json:"preflight_connection"`
```

Comment:
Wo we need a additional flag for that ? Can we use the caller by   PeerCert and check if the call was done via WebUI or tsh app access ?

Reply:
Yes, we could rely on the requested certificate. However, since it is an API endpoint, my thought is that having this behavior set explicitly is better than assuming it based on caller characteristics. Rather than that, I don't mind changing it to be the default behavior for non-certificate requests.

Reply:
We cannot use the `HasClientCert` and `HasSession` functions because they rely on application session credentials, which are not present here. They become available only after the application launcher is executed on the WebUI or after this API endpoint is called on the CLI.